paddle/fluid/imperative/py_layer_fwd.h

```diff
@@ -78,28 +89,34 @@ py::object PyLayerApply(const platform::Place& place, const py::handle& cls,
   // process args,`input_vars` only collect `imperative::VarBase`
   if (!args.empty()) {
     for (auto ptr = args.begin(); ptr != args.end(); ptr++) {
-      try {
-        if (Py_None != ptr->ptr()) {
+      // Only collect Tensor type in 'args' and pass them to backward. Ignore
+      // other types of input temporarily.
+      if (py::isinstance<imperative::VarBase>(*ptr)) {
+        try {
           auto a = ptr->cast<std::shared_ptr<VarBase>>();
           input_vars.push_back(a);
+        } catch (py::cast_error& err) {
+          PADDLE_THROW(platform::errors::InvalidArgument(
+              "`%s` can not be cast into `Tensor`.",
```

Comment:
give more specific error message here?

`The PyLayer.forward function contains invalid argument, the `%s` type argument can not be cast into `Tensor`.`?

same below

Reply:
Done, thx.